src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -5403,6 +5403,140 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+   /**
+   *  Arguments:
+   *
+   *  Input:
+   *  c_rarg0   - src_start
+   *  c_rarg1   - src_offset
+   *  c_rarg2   - src_length
+   *  c_rarg3   - dest_start
+   *  c_rarg4   - dest_offset
+   *  c_rarg5   - isURL
+   *
+   */
+  address generate_base64_encodeBlock() {
+
+    static const char toBase64[64] = {
+      'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+      'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+      'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+      'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+      '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '+', '/'
+    };
+
+    static const char toBase64URL[64] = {
+      'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+      'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+      'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+      'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+      '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '-', '_'
+    };
+
+    __ align(CodeEntryAlignment);
+    StubCodeMark mark(this, "StubRoutines", "encodeBlock");
+    address start = __ pc();
+
+    Register src   = c_rarg0;  // source array
+    Register sp    = c_rarg1;  // source start offset
+    Register sl    = c_rarg2;  // source end offset
+    Register dst   = c_rarg3;  // dest array
+    Register dp    = c_rarg4;  // position for writing to dest array
+    Register isURL = c_rarg5;  // Base64 or URL chracter set
+
+    Register codec  = rscratch1;
+    Register length = rscratch2;
+
+    Label ProcessData, Process48B, Process24B, Process3B, SIMDExit, Exit;
+
+    #define BASE64_ENCODE_SIMD_ROUND(in0, in1, in2, out0, out1, out2, out3, SZ) \
+      __ ld3(in0,  in1, in2, __ T##SZ##B, __ post(src, 3 * SZ));                \
```

Comment:
There's no need for this to be a macro -- as far as I can see.

Reply:
Thanks for the suggestions.

Just updated a version.
The register name `sp` is changed to `soff`, and the macro is unpacked into code block `Process48B` and `Process24B`.

Verified with `test/jdk/java/util/Base64/`.

Reply:
I'm sorry, there's no way I wanted the macro to be unpacked; I wanted it to be a function. I apologize for not being clear.

Reply:
Okay.

Updated, reconstructed the macro as function `generate_base64_encode_simdround`.

Thanks.